lis/dataassim/obs/SMOS_L2sm/SMOSL2sm_Mod.F90

```diff
@@ -205,9 +207,10 @@ subroutine SMOSL2sm_setup(k, OBS_State, OBS_Pert_State)
        call ESMF_AttributeSet(OBS_State(n),"Data Assimilate Status",&
             .false., rc=status)
        call LIS_verify(status)
-       
+
+      !SM09012024 change LIS_rc%ngrid(n) with LIS_rc%obs_ngrid(k) 
        call ESMF_AttributeSet(OBS_State(n),"Number Of Observations",&
-            LIS_rc%ngrid(n),rc=status)
+            LIS_rc%obs_ngrid(k),rc=status)
        call LIS_verify(status)
```

Comment:
I don't understand exactly what this is doing: there is a loop over n->nnest, and obs_ngrid(k) is overwritten for all nnest?

Reply:
(k) is assimilation instance; in line with SMAP DA. The old SMOS DA code was not functioning.